### ISSUE TITLE
博主好，推理结果长度较短，且内容大量重复怎么解决？

### ISSUE THREAD

**Issue by Hoogck** [RowitZou/DAMS#1]
下面是我训练出来的模型，在测试集上的推理结果，可以看到生成摘要较短，而且存在大量重复内容：
1	stanislaw is not going to the ball . <q> she already did . <q> he will be back alreads . <q> taylor will be there in the fireworks last last night . <q> she
2	stanislaw is not going to the ball . <q> he will be home arrives . <q> she already will be back alreads . <q> taylor is be back arrive soon . <q> he
3	stanislaw is not going to the ball . <q> she already did . <q> he will be back alreads . <q> taylor will be there in 15 minute lesson . <q> taylor
4	stanislaw is not going to the ball . <q> she already dinner . <q> he isabababed for the fireworks last last night . <q> taylor will be back alreads to the pool in the ball is broked alreadd . <q> she
5	stanislaw is not going to the ball . <q> she already did . <q> he is beavers in the fireworks last last night . <q> taylor is be back almost stanid . <q>

数据格式已经转换跟您网盘提供的一致，请问这个是什么原因，可以怎么解决呢？敬谢~

**Comment by RowitZou** [comment]
可以提供训练log或者测试的rouge结果吗

**Comment by Hoogck** [comment]
1.预训练的acc还蛮高的:
[2021-11-17 02:10:55,214 INFO] Step 73995/80000; acc:  92.98; ppl:  1.35; xent: 0.30; rlloss: 0.00; tploss: 0.00; lr: 0.00000368;   0/109 tok/s;  58420 sec
[2021-11-17 02:10:58,909 INFO] Step 74000/80000; acc:  95.79; ppl:  1.21; xent: 0.19; rlloss: 0.00; tploss: 0.00; lr: 0.00000368;   0/125 tok/s;  58424 sec

2.fine-tune阶段的acc一直上不去：
[2021-11-18 09:04:53,684 INFO] Step 4285/30000; acc:  40.94; ppl: 62.85; xent: 4.14; rlloss: -0.44; tploss: 0.00; lr: 0.00000012;   0/ 26 tok/s;  84530 sec
[2021-11-18 09:06:52,045 INFO] Step 4290/30000; acc:  44.10; ppl: 33.02; xent: 3.50; rlloss: 0.19; tploss: 0.00; lr: 0.00000012;   0/ 

3.生成的结果：
The index:25905,The pred_summ:***客户来电称自己的手机号码信号是正常的，但是却无法接听电话，请后台处理
The index:25906,The pred_summ:用户来电对本月产生的流量费不认可，我方按照上网日志解释用户不认可。称没有使用过，要求我方处理费用。烦请处理，谢谢！
The index:25907,The pred_summ:********用户来电称对手机产生的欠费不认可，要求核实原因，请处理，谢谢！
The index:25908,The pred_summ:用户来电反映称对1月份产生的20元流量费有疑义，我方经系统查询按上网日志向用户解释，用户不认可要求核实原因，请处理，谢谢。

**Comment by RowitZou** [comment]
目前我使用的预训练数据集均是英文，是无法直接在中文数据集上 finetune 的。可能需要使用中文的对应语料进行预训练。
另外，预训练时有两个loss，分别为 advenc loss 以及 advdec loss，您提供的log版本显示，使用的代码似乎是较早的版本，希望您能再次确认一下。

**Comment by Hoogck** [comment]
好的，感谢大佬，我转了一批中文语料进行预训练了的。请问我怎么可以让生成的语句更长一点，或者更多的拷贝忠于原文内容呢？

**Comment by RowitZou** [comment]
看您的需求是客服场景对话，可以参考我的上一篇工作的代码：https://github.com/RowitZou/topic-dialog-summ。 

基本思想是先从长对话抽取关键句，再对关键句进行refine。本质上是对 FastRL (2018) 论文的改进版本，而我们使用了预训练模型将其复现，并加入了主题模型。对于具体的建议，希望您在上述 project 提 issue，我再进行详细解答。

**Comment by Hoogck** [comment]
好的，再次感谢博主；那我去您说的那个project里面提issue。
